packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -547,6 +547,14 @@ function transform(codeGen: CodeGen): t.Expression {
                         const styleAST = styleMapToStyleDeclsAST(styleMap);
                         data.push(t.property(t.identifier('styleDecls'), styleAST));
                     }
+                } else if (name === 'slot') {
+                    let slotValue;
+                    if (isExpression(value)) {
+                        slotValue = codeGen.bindExpression(value);
+                    } else {
+                        slotValue = isStringLiteral(value) ? t.literal(value.value) : t.literal('');
```

Comment:
Is there any way this could be a breaking change? E.g. are we treating `slot={boolean}` or `slot={null}` or `slot={undefined}` or any other non-string values any different from before?